File: JavaClient/src/main/java/com/security/crypto/KeyManager/Keys.java
package com.security.crypto.KeyManager;


public abstract class Keys {

    private final Type type;

    public enum Type {

        SymetricKeyGenerator, DHSecretKey, DHIntegrityKey
    }

    //this constructor  doesn't actually "BUILD" the object, it is used to initialize fields.
    public Keys(Type type) {
        this.type = type;
    }

    public Boolean isType(Type type) {
        return (this.type == type);
    }


    public abstract String RetriveSessionKey();
}

File: SecureUWPClient/SecureUWPClient/KeyManager/Keys.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SecureUWPClient.KeyManager
{
    public abstract class Keys
    {

        private Type type;

       public enum Type
        {

            DHCipherKey, DHIntegrityKey
        }


        public Keys(Type type)
        {
            this.type = type;
        }

        public Boolean isType(Type type)
        {
            return (this.type == type);
        }


        public abstract String RetriveSessionKey();
    }
}
